activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -635,12 +643,11 @@ def default_index_type?(index) # :nodoc:
       end
 
       def build_insert_sql(insert) # :nodoc:
-        sql = +"INSERT #{insert.into} #{insert.values_list}"
+        sql = +"INSERT"
+        sql << " IGNORE" if insert.skip_duplicates?
+        sql << " #{insert.into} #{insert.values_list}"
 
-        if insert.skip_duplicates?
```

Comment:
Switching this to `INSERT IGNORE` 👌

Reply:
@adrianna-chang-shopify @fatkodima @nvasilevski Why is this change desirable? I don't see any discussion of why we want to use `INSERT IGNORE` over a no-op `ON DUPLICATE KEY UPDATE`. I haven't tested but I think the behaviour may be different.

Reply:
`INSERT IGNORE` generates a warning and `ON DUPLICATE KEY UPPDATE` does not.

Reply:
Ok, so I thought that these are equivalent. Do you mean MariaDB warning?

Reply:
Oh, I see from the docs what you mean. Thanks for pinging! I will open a PR with fix.

Reply:
With RETURNING clause INSERT ON DUPLICATE UPDATE no_op always returns all (even not actually inserted) rows in result set.
INSERT IGNORE returns only the actually inserted rows and in this behaviour it coincides with postgresql adapter.

```sql
MariaDB [db]> CREATE TABLE tmp_users (
    ->   id int(11) NOT NULL AUTO_INCREMENT,
    ->   email varchar(255) NOT NULL,
    ->   PRIMARY KEY (id),
    ->   UNIQUE KEY unique_users_email (email)
    -> );
Query OK, 0 rows affected (0,010 sec)

MariaDB [db]> insert ignore tmp_users (email) values ('one@example.com'), ('two@examp.com');
Query OK, 2 rows affected (0,002 sec)
Records: 2  Duplicates: 0  Warnings: 0

MariaDB [db]> insert tmp_users (email) values ('one@example.com'), ('two@examp.com') ON DUPLICATE KEY UPDATE id=id returning id, email;
+----+-----------------+
| id | email           |
+----+-----------------+
|  1 | one@example.com |
|  2 | two@examp.com   |
+----+-----------------+
2 rows in set (0,001 sec)

MariaDB [db]> insert ignore tmp_users (email) values ('one@example.com'), ('two@examp.com') returning id, email;
Empty set, 2 warnings (0,001 sec)
```

After https://github.com/rails/rails/pull/49890 we won't know which of the rows were inserted.

Reply:
@jhawthorn , @fatkodima 👀 ☝🏻

Reply:
Can you open a new issue with all of the information describing your problem? Pinging maintainers is not a good way to get a response

Reply:
I think this was addressed already: https://github.com/rails/rails/commit/eecd67236039bc0fe67865521d33e05d3ff81c19